cirq/study/trial_result.py

```diff
@@ -246,7 +259,7 @@ def _repr_pretty_(self, p: Any, cycle: bool) -> None:
             p.text(str(self))
 
     def __str__(self):
-        return _keyed_repeated_bitstrings(self.measurements)
+        return _keyed_repeated_bitstrings(self.data)
 
     def _value_equality_values_(self):
-        return self.measurements, self.repetitions, self.params
+        return repr(self.data), self.params
```

Comment:
Are you sure pandas guarantees the repr contains all information?

Reply:
Going by https://docs.python.org/3/reference/datamodel.html#object.__repr__, it ideally should. Any better suggestions?
Given that the DataFrame is unhashable and so are the Series elements inside our DataFrame, short of iterating through all the measured bits, this is the best option IMO.
The previous method of using self.measurements directly was buggy because it couldn't work for multi-element input arrays and the tests didn't catch that. Modified the test too.